vars/buildPlugin.groovy

```diff
@@ -34,47 +36,73 @@ def call(Map params = [:]) {
                             else {
                                 error 'buildPlugin must be used as part of a Multibranch Pipeline *or* a `repo` argument must be provided'
                             }
+
+                            isMaven = fileExists('pom.xml')
                         }
 
                         stage("Build (${stageIdentifier})") {
-                            List<String> mavenOptions = [
-                                '--batch-mode',
-                                '--errors',
-                                '--update-snapshots',
-                                '-Dmaven.test.failure.ignore=true',
-                                "-DskipAfterFailureCount=${failFast}",
+                            String jdkTool = "jdk${jdk}"
+                            List<String> env = [
+                                    "JAVA_HOME=${tool jdkTool}",
+                                    'PATH+JAVA=${JAVA_HOME}/bin',
                             ]
-                            if (jenkinsVersion) {
-                                mavenOptions += "-Djenkins.version=${jenkinsVersion}"
+                            String command
+                            if (isMaven) {
+                                List<String> mavenOptions = [
+                                        '--batch-mode',
+                                        '--errors',
+                                        '--update-snapshots',
+                                        '-Dmaven.test.failure.ignore=true',
+                                        "-DskipAfterFailureCount=${failFast}",
+                                ]
+                                if (jenkinsVersion) {
+                                    mavenOptions += "-Djenkins.version=${jenkinsVersion}"
+                                }
+                                command = "mvn ${mavenOptions.join(' ')} clean install"
+                                env << "PATH+MAVEN=${tool 'mvn'}/bin"
+                            } else {
+                                List<String> gradleOptions = [
+                                        '--no-daemon',
+                                        'cleanTest',
+                                        'build',
+                                ]
+                                command = "gradlew ${gradleOptions.join(' ')}"
+                                if (isUnix()) {
+                                    command = "./" + command
+                                }
```

Comment:
Good point! But in any case, sorry — I misread.  This is just adding the `./` prefix for Unix, not actually *running* the command…